### DISCUSSION TITLE
Sorry i am extremly new to this how do i use this?

### DISCUSSION THREAD

**Discussion by jjygs** [Official-Husko/NN-Downloader#41] [Q&A] [answered]
i downloaded the programs? and ran them but it asks me for an api key?
really have 0 clue on what im doing or how to use so please explain if possible
![Screenshot 2024-07-19 044255](https://github.com/user-attachments/assets/7169e974-46c1-4681-b302-d9441ecbb521)

**Comment by Official-Husko** [comment]
First of all what sites would you like to download from? certain sites require an api key like all E-Sites so E621, E926 etc. You can obtain these from the website itself if you click on account and then on "Manage API Access" and in there you can generate a key.

This key is used to interact with the website and get the data of posts. You can set this api key up in the config.json file using notepad here:

```json
        "e621": {
            "apiUser": "",
            "apiKey": ""
        },
```

apiUser is your E621 Username and apiKey is the key you got from the website. 


If you need further assistance feel free to let me know and i will help you out.

**Reply by jjygs** [reply]
Thank you very much for this explanation i will try this out right now

**Reply by jjygs** [reply]
Hello again, ok so i did as you said and i was able to download some pages from an artist on e621 but it did not seem to download all the pages? even tho i left the amount of pages blank to download them all. Did i have to put the specific amount of pages to download still or something?

**Reply by Official-Husko** [reply]
> Hello again, ok so i did as you said and i was able to download some pages from an artist on e621 but it did not seem to download all the pages? even tho i left the amount of pages blank to download them all. Did i have to put the specific amount of pages to download still or something?

It should just work out of the box. How many images were downloaded?

**Reply by jjygs** [reply]
So the first time 222 images were downloaded
Second i tried specifying number of pages (9) and 442 images were downloaded

on e621 the artist has a total of 664 images across 9 pages

**Reply by Official-Husko** [reply]
> So the first time 222 images were downloaded Second i tried specifying number of pages (9) and 442 images were downloaded
> 
> on e621 the artist has a total of 664 images across 9 pages

Weird, the artist is `kiseff` correct? I will give it a try myself to see if i can recreate the issue.

**Reply by Official-Husko** [reply]
> So the first time 222 images were downloaded Second i tried specifying number of pages (9) and 442 images were downloaded
> 
> on e621 the artist has a total of 664 images across 9 pages

It seems fine for me. Try this, go to the e621 site and go to your settings and check if "safe mode" is disabled and check the blacklisted tags and remove them for testing. After doing that save the changes and try again.

Edit: I got exactly 664 images after letting it run for a few minutes with kiseff as the tag. I have uploaded the full zip with all the images since i already have them now but please still try the fix above. [GoFile](https://gofile.io/d/bH57gp)

**Reply by jjygs** [reply]
I tried doing what you said but this is what happend
![Screenshot 2024-07-23 165447](https://github.com/user-attachments/assets/0e706bba-e1d7-4418-90c0-86d2d5540813)

**Reply by jjygs** [reply]
no idea why it doesnt work, should i screen shot the config notepad too? but blur my api?

**Reply by Official-Husko** [reply]
> no idea why it doesnt work, should i screen shot the config notepad too? but blur my api?

Yes please. also if possible show me the settings page on e621 with the blacklisted tags and all that.

**Reply by jjygs** [reply]
Ok here ty, il also put the run time too if that helps.
![Screenshot 2024-07-24 025347](https://github.com/user-attachments/assets/8f0b48ce-1ed6-4f4f-9088-90bcad517f5a)
![Screenshot 2024-0
[runtime.log](https://github.com/user-attachments/files/16356079/runtime.log)
7-24 025241](https://github.com/user-attachments/assets/4397a5d0-783f-42de-96a7-11b65c81d6a7)

**Reply by jjygs** [reply]
[runtime.log](https://github.com/user-attachments/files/16356083/runtime.log)

**Reply by Official-Husko** [reply]
> [runtime.log](https://github.com/user-attachments/files/16356083/runtime.log)

I'm sorry I'm quite busy right now. i will respond once i got some time.

**Reply by Official-Husko** [reply]
the last thing i can think of is try it without the proxies. Disable them and try to download the images again and make sure to delete any databases in the db folder. e621.db for example

**Reply by jjygs** [reply]
Ahh i apperciate all the help still, i set proxies to false in config  and deleted the e621 db 
![Screenshot 2024-07-26 104813](https://github.com/user-attachments/assets/d1e8d866-88a1-4176-9d29-e3134f093f04)
Looking at it, it seems its going to download 320 again tho :c

**Reply by jjygs** [reply, marked as answer]
O wait, it downloaded it all now, maybe it was the proxie after all, thank you!
